### ISSUE TITLE
`maxiter` parameter

### ISSUE THREAD

**Issue by cortner** [JuliaMath/IterativeSolvers.jl#168]
```
using IterativeSolvers
L1d(N) = spdiagm((-ones(N-1), 2*ones(N), -ones(N-1)), (-1,0,1), N, N) * N^2
function L2d(N)
  L1 = L1d(N)
  A = kron(speye(N), L1) + kron(L1, speye(N))
  b = ones(size(A,1))
  return A, b
end
A, b = L2d(100)
TOL = 1e-4
x = IterativeSolvers.cg(A, b; tol=1e-4)
norm(A * x - b, Inf)
```
The output of the final line is `35.59...` 

This is because `maxiter = size(A,2)` instead of `maxiter = size(A,2)^2` as I think it should be (and indeed as is claimed in the documentation).  I believe this must have changed recently. The above problem is just a finite-difference laplacian on a square domain. Something as basic as that should not fail with default parameters.

**Comment by andreasnoack** [comment]
@haampie Any thoughts?

**Comment by haampie** [comment]
`n^2` is way too large for n by n matrices I would think. It's pessimistic at the very least.

In exact arithmetic `n` is the worst-case number of iterations. If the matrix is well-conditioned (maybe via a preconditioner) and the tolerance is like the above, you can expect less than `n` iterations.

The default value is `maxiter = min(20, size(A, 1))` at the moment, but it is not documented. It's the same as MATLABs `pcg` function. The arbitrary value `20` is there just to make sure CG quits early. Maybe we can just set `maxiter = size(A, 2)`, since that is actually documented.

```julia
x, history = cg(A, b; tol = 1e-4, log = true)
@show history.iters
@show history.isconverged
```
shows 

```
history.iters = 20
history.isconverged = false
```

And if you specify `maxiter` to be the size of A, you get

```julia
x, history = cg(A, b; tol = 1e-4, log = true, maxiter = 10_000)
@show history.iters
@show history.isconverged
```

```
history.iters = 134
history.isconverged = true
```

So 134 iterations is enough to reduce the error in the A-norm by 1e-4 starting with x =0. Since n = 10_000, taking `maxiter = 100_000_000` is quite pessimistic ^^.

TL;DR: Current default `maxiter` is not documented and kept low by default as in MATLAB (but yeah, it's arbitrary)

**Comment by cortner** [comment]
sorry, the n^2 was a brain-burp due to the 2D problem. it should be n of course. so the `20` is the problem. This seems extremely low to me. If you don't know what you are doing, then is it not more confusing that you get garbage results? If you really want to keep `20` then how about printing a warning that the iteration did not converge?

**Comment by haampie** [comment]
Agreed, `n` is arguably a better default. Not sure about printing a warning (see also #6). Setting `log = true` already gives you convergence info.
